src/main/kotlin/com/terraformation/backend/tracking/db/PlantingSiteStore.kt

```diff
@@ -1328,66 +1328,35 @@ class PlantingSiteStore(
   /**
    * Makes a monitoring plot unavailable for inclusion in future observations.
    *
-   * The requested plot's "is available" flag is set to false.
+   * The requested plot's "is available" flag is set to false, and its permanent cluster number is
+   * cleared. If the plot was permanent, a new one with the removed cluster number will be created
+   * next time [ensurePermanentClustersExist] is called.
    *
-   * If the requested plot is part of a permanent cluster, the cluster is destroyed: the remaining
-   * plots in the cluster are updated such that they're no longer in a permanent cluster at all (but
-   * are still available for selection as temporary plots). A new cluster with the removed cluster
-   * number will be created next time [ensurePermanentClustersExist] is called.
-   *
-   * @return The plots that were modified. If the requested plot was part of a permanent cluster,
-   *   the "added plots" property will be the IDs of the plots in the cluster that was swapped in to
-   *   replace the original cluster (or an empty list if there wasn't a replacement cluster
-   *   available), and the "removed plots" property will be the IDs of the plots in the same cluster
-   *   as the requested plot. If the requested plot wasn't part of a permanent cluster, the "added
-   *   plots" property will be empty and the "removed plots" property will only include the
-   *   requested plot.
+   * @return The plots that were modified. This is always either an empty list (if the plot was
+   *   already unavailable) or a removed plots list with just the requested plot (if the plot was
+   *   available before and we've just marked it as unavailable).
    */
   fun makePlotUnavailable(monitoringPlotId: MonitoringPlotId): ReplacementResult {
     val plotsRow =
         monitoringPlotsDao.fetchOneById(monitoringPlotId)
             ?: throw PlotNotFoundException(monitoringPlotId)
 
-    if (plotsRow.isAvailable == false) {
-      return ReplacementResult(emptySet(), emptySet())
-    }
-
-    val subzonesRow =
-        plantingSubzonesDao.fetchOneById(plotsRow.plantingSubzoneId!!)
-            ?: throw PlantingSubzoneNotFoundException(plotsRow.plantingSubzoneId!!)
-    val permanentCluster = plotsRow.permanentCluster
-    val plantingZoneId = subzonesRow.plantingZoneId!!
-
-    requirePermissions { updatePlantingSite(subzonesRow.plantingSiteId!!) }
+    requirePermissions { updatePlantingSite(plotsRow.plantingSiteId!!) }
 
-    return dslContext.transactionResult { _ ->
+    return if (plotsRow.isAvailable == true) {
```

Comment:
NIT - could change this to just `if (plotsRow.isAvailable)` now

Reply:
The `== true` is because `isAvailable` is nullable (because all the fields on the `Row` classes are nullable). The alternative would be `if (plotsRow.isAvailable!!)` but IntelliJ suggests changing that to `== true`.